packages/broker/test/integration/plugins/operator/OperatorValueBreachWatcher.test.ts

```diff
@@ -0,0 +1,121 @@
+import { generateWalletWithGasAndTokens } from './smartContractUtils'
+
+import { OperatorValueBreachWatcher } from '../../../../src/plugins/operator/OperatorValueBreachWatcher'
+import { STREAMR_DOCKER_DEV_HOST, createClient, createTestStream } from '../../../utils'
+
+const fastChainConfig = config.dev2
+const theGraphUrl = `http://${STREAMR_DOCKER_DEV_HOST}:8800/subgraphs/name/streamr-dev/network-subgraphs`
+
+const logger = new Logger(module)
+
+const STREAM_CREATION_KEY = '0xb1abdb742d3924a45b0a54f780f0f21b9d9283b231a0a0b35ce5e455fa5375e7'
+
+async function getTotalUnwithdrawnEarnings(operatorContract: Operator): Promise<bigint> {
+    const { earnings } = await operatorContract.getEarningsFromSponsorships()
+    let unwithdrawnEarnings = BigInt(0)
+    for (const e of earnings) {
+        unwithdrawnEarnings += e.toBigInt()
+    }
+    logger.debug(`Total unwithdrawn earnings: ${formatEther(unwithdrawnEarnings.toString())} (t = ${Date.now()})`)
+    return unwithdrawnEarnings
+}
+
+describe('OperatorValueBreachWatcher', () => {
+    let provider: Provider
+    let token: TestToken
+    let streamId: string
+
+    const deployNewOperator = async () => {
+        const operatorWallet = await generateWalletWithGasAndTokens(provider, fastChainConfig)
+        logger.debug('Deploying operator contract')
+        const operatorContract = await deployOperatorContract(fastChainConfig, operatorWallet, { operatorSharePercent: 10 })
+        logger.debug(`Operator deployed at ${operatorContract.address}`)
+        const operatorConfig = {
+            operatorContractAddress: toEthereumAddress(operatorContract.address),
+            provider,
+            theGraphUrl,
+            signer: operatorWallet,
+            maxSponsorshipsCount: 20,
+            minSponsorshipEarnings: 1 // full tokens
+        }
+        return { operatorWallet, operatorContract, operatorConfig }
+    }
+
+    beforeAll(async () => {
+        provider = new JsonRpcProvider(`http://${STREAMR_DOCKER_DEV_HOST}:8547`)
+        logger.debug('Connected to: ', await provider.getNetwork())
+
+        logger.debug('Creating stream for the test')
+        const client = createClient(STREAM_CREATION_KEY)
+        streamId = (await createTestStream(client, module)).id
+        await client.destroy()
+
+        token = new Contract(fastChainConfig.contracts.DATA, tokenABI) as unknown as TestToken
+    }, 60 * 1000)
+
+    it('can find a random operator, excluding himself', async () => {
+        const { operatorContract, operatorConfig } = await deployNewOperator()
+        // deploy another operator to make sure there are at least 2 operators
+        await deployNewOperator()
+
+        const operatorValueBreachWatcher = new OperatorValueBreachWatcher(operatorConfig)
+        const randomOperatorAddress = await operatorValueBreachWatcher.helper.getRandomOperator()
+        if (randomOperatorAddress === undefined) {
+            throw new Error('No random operator found')
+        }
```

Comment:
```suggestion
        expect(randomOperatorAddress).toBeDefined()
```

Reply:
one of the points of having that comparison is making it clear to TypeScript that `randomOperatorAddress` can not be undefined. TS can't infer it from `expect`.

Reply:
Left it as-is for now